src/OnlineRetailPortal.Core/Models/ProductList.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OnlineRetailPortal.Core
+{
+    public class ProductList
+    {
+        public string Id { get; set; }
+        public string Name { get; set; }
+        public Price Price { get; set; }
+        public Price Currency { get; set; }
```

Comment:
again Currency can not be of type Price.

Reply:
currency is a property of Price, and this property is type of Currency.